typescript/test/unit/baseline.ts

```diff
@@ -79,10 +78,57 @@ describe('CodeGeneratorTest', () => {
           `-I${PROTOS_DIR} ` +
           ECHO_PROTO_FILE
       );
-      assert.strictEqual(
-        fs.readFileSync(GENERATED_CLIENT_FILE).toString(),
-        fs.readFileSync(CLIENT_LIBRARY_BASELINE).toString()
-      );
+      const outputFiles = fs.readdirSync(OUTPUT_DIR);
+      // store every item (file or directory with full path in output dir and baseline dir) in stack, pop once a time
+      const protoItemStack: Item[] = [];
+      outputFiles.forEach(file => {
+        const fileFullPath = path.join(OUTPUT_DIR, file);
+        const baselinePath = path.join(BASELINE_DIR, file);
+        protoItemStack.push(new Item(file, fileFullPath, baselinePath));
+      });
+      console.warn(protoItemStack);
+      while (protoItemStack.length !== 0) {
+        const item = protoItemStack.pop();
+        if (!item) continue;
+        // if item is a file, compare it with baseline
+        if (fs.lstatSync(item.outputPath).isFile()) {
+          checkIdenticalFile(
+            item.outputPath,
+            item.baselinePath + BASELINE_EXTENSION
+          );
+        } else {
+          // if item is a directory, loop the folder and put every item in stack again
+          const items2level = fs.readdirSync(item.outputPath);
+          items2level.forEach(item2level => {
+            const fileFullPath2level = path.join(item.outputPath, item2level);
```

Comment:
style: these three lines are very similar to the three lines above (lines 85-87).
Maybe moving the `path.join` logic to the `Item` constructor can simplify this code a little bit.